projects/fal/src/fal/toolkit/file/file.py

```diff
@@ -79,15 +89,27 @@ def __init__(self, **kwargs):
                     "file_size": len(data.data),
                 }
             )
+        else:
+            data = None
 
         super().__init__(**kwargs)
         if data is not None:
             self._file_data = data
 
     # Pydantic custom validator for input type conversion
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.__convert_from_str
+    if IS_PYDANTIC_V2:
+        @classmethod
+        def __get_pydantic_core_schema__(
+            cls, source_type: Any, handler: GetCoreSchemaHandler
+        ) -> CoreSchema:
+            return core_schema.no_info_before_validator_function(
+                cls.__convert_from_str,
+                handler(source_type),
+            )
+    else:
+        @classmethod
+        def __get_validators__(cls):
+            yield cls.__convert_from_str  
```

Comment:
Ray just uses pydantic.v1 https://github.com/ray-project/ray/blob/9cb1dc9e682a087a32f47838fa02ca35f9b1b6ba/python/ray/_private/pydantic_compat.py#L57 for compat layer, but I properly migrated out class here because it is more proper long term and also allows us to use these in testing v1 and v2 natively.